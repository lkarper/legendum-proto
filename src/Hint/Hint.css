.Hint__toggle-show-add {
    margin: 5px;
}