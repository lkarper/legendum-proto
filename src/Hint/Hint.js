import React, { useState, useContext } from 'react';
import UserContext from '../contexts/UserContext';
import TokenService from '../services/token-service';
import NotesService from '../services/notes-service';
import SaveHint from '../SaveHint/SaveHint';
import './Hint.css';

const Hint = (props) => {
    const { hint } = props;

    const context = useContext(UserContext);
    
    const [showAdd, toggleShowAdd] = useState(false);
    const [noteAdded, setNoteAdded] = useState(false);
    const [apiError, setApiError] = useState(false);
    
    const handleSaveHint = (event, customNote) => {
        event.preventDefault();
        NotesService.postNote(hint.id, customNote)
            .then(note => {
                context.addNote(note);
                setNoteAdded(true);
                setApiError(false);
            })
            .catch(error => {
                console.log('error', error);
                setApiError(true);
            });
    }

    return (
        <li>
            <p className='Hint__hint-text'>{hint.hint}</p>
            {TokenService.hasAuthToken() &&
                <div className='Hint__add-hint-container'>
                    <button
                        className='Hint__toggle-show-add button' 
                        onClick={() => toggleShowAdd(!showAdd)}
                        disabled={noteAdded}    
                    >
                        {showAdd ? 'Nevermind' : 'Add Hint to Journal'}
                    </button>
                    {(showAdd && !noteAdded) && <SaveHint onSubmitHint={handleSaveHint} />}
                    {apiError && <p>Error: Could not save note. Check your connection and try again.</p>}
                    {noteAdded && <p>Note saved successfully!</p>}
                </div> 
            }
        </li>
    );
}

export default Hint;
