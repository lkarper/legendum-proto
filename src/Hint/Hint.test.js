import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Hint from './Hint';

describe('Hint component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Hint />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<Hint />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
