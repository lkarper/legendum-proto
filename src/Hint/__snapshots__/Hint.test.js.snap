// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hint component renders the UI as expected 1`] = `
<Hint
  hint={Object {}}
>
  <li>
    <p>
      Error: Looks like something went wrong. Check your connection and try again.
    </p>
  </li>
</Hint>
`;
